src/Microsoft.AspNetCore.Sockets.Client.Http/HttpConnection.cs

```diff
@@ -123,13 +123,12 @@ private Task StartAsyncCore()
             return _startTcs.Task;
         }
 
-        private async Task StartAsyncInternal()
+        private async Task StartAsyncInternal(CancellationToken cancellationToken)
```

Comment:
This cancellation token seems only to be used for the negotiate request. How about cancelling starting the transport?

Reply:
Adding it to the the transports StartAsync methods